extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,78 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationLinks();
+		$hasMobileApplicationBadges = count($mobileApplicationsLinks) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl( $this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl( $this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationLinks() {
+		$result = [];
+
+		$response = $this->fetchMobileApplications();
+		$siteId = $this->wg->CityId;
+
+		if ( $this->applicationsExistFor($siteId, $response ) ) {
+			$json = json_decode($response, true);
+
+			foreach ( $json['apps'] as &$app ) {
+				foreach ( $app['languages'] as &$language ) {
+					if ( $language['wikia_id'] == $siteId ) {
+						if ( $app['android_release'] ) {
+							$release = $app['android_release'];
+							$result['android'] = "https://play.google.com/store/apps/details?id=$release";
+						}
+						if ( $app['ios_release'] ) {
+							$release = $app['ios_release'];
+							$result['ios'] = "https://itunes.apple.com/us/app/id$release";
+						}
+						break;
+					}
+				}
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * @param string siteId - current site id
+	 * @param string response - raw, not parsed response
+	 *
+	 * @return boolean
+	 */
+	private function applicationsExistFor( $siteId, $response ) {
+		return strpos( $response, 'wikia_id":' . $siteId ) !== false;
+	}
+
+	/**
+	 * @return string
+	 */
+	private function fetchMobileApplications() {
+		// currently it does not matter if Android or iOS value is added, data is returned for both Android and iOS
+		return Http::request( "GET",  'https://services.wikia.com/mobile-applications/platform/android' );
+	}
+
 	/**
 	 * TODO Move this into unsubscribe extension?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)